snippets/price.liquid

```diff
@@ -37,94 +38,97 @@
   endif
 -%}
 
-{%- unless target == nil -%}
-<div
-  class="
-    price
-    {%- if price_class %} {{ price_class }}{% endif -%}
-    {%- if available == false %} price--sold-out{% endif -%}
-    {%- if compare_at_price > price and product.quantity_price_breaks_configured? != true %} price--on-sale{% endif -%}
-    {%- if compare_at_price > price and product.quantity_price_breaks_configured? %} volume-pricing--sale-badge{% endif -%}
-    {%- if product.price_varies == false and product.compare_at_price_varies %} price--no-compare{% endif -%}
-    {%- if show_badges %} price--show-badge{% endif -%}
-  "
->
-  <div class="price__container">
-    {%- comment -%}
-      Explanation of description list:
-        - div.price__regular: Displayed when there are no variants on sale
-        - div.price__sale: Displayed when a variant is a sale
-    {%- endcomment -%}
-    <div class="price__regular">
-      {%- if product.quantity_price_breaks_configured? -%}
-        {%- if show_compare_at_price and compare_at_price -%}
-          {%- unless product.price_varies == false and product.compare_at_price_varies %}
-            <span class="visually-hidden visually-hidden--inline">
-              {{- 'products.product.price.regular_price' | t -}}
-            </span>
-            <span>
-              <s class="price-item price-item--regular variant-item__old-price">
-                {% if settings.currency_code_enabled %}
-                  {{ compare_at_price | money_with_currency }}
-                {% else %}
-                  {{ compare_at_price | money }}
-                {% endif %}
-              </s>
-            </span>
-          {%- endunless -%}
+{%- unless target == null and placeholder == null -%}
```

Comment:
One thing I actually noticed which is minimal. But curious to know how we want to go about it. 

There is an issue where if you add a feat. collection to the product page, the placeholder product cards get a price that matches the currently selected variant. 
So with the `placeholder` prop I added, we can actually check if it is true and assign `null` to the `target` to keep our default `19.99` price as expected. 

So I might keep the `placeholder` approach for that reason 🤔 

The other solution could be to keep your approach with `product != null` but change it so it's not referring to the actual `product` in the snippet props. Where right now it says in the comment at the top of the file:
```
- product: {Object} Product Liquid object (optional)
```
Cause when on a product template, it will always return the product object of the template as a fallback. 
It should maybe be named instead `product_resource` so we can actually know whether that prop was passed when rendering the snippet or not 🤔 
And the file also points to `product.` a few times instead of `target.`

[Recording](https://screenshot.click/25-45-llp2s-hs7y0.mp4)

Let me know what you think. cc: @Roi-Arthur @emma-boardman @lhoffbeck

Reply:
Yep I can see the theoretical improvement as it preserves the same example price 👍🏼

On the other hand, I think the important piece is to show _any_ price (rather than none), and ultimately the example shouldn't stick around for too long. I also somewhat doubt that Merchants would go around changing the product to see how it affects the example.
And lastly, if we're going back to the initial point that this messes up the AI generated themes, I think that this is mostly relevant to the front page.

Overall I think I would go for the least amount of code/complexity but happy to go either way.